tests/sentry/incidents/test_logic.py

```diff
@@ -1569,6 +1569,151 @@ def test_update_infer_detection_type(self, mock_seer_request):
 
         assert updated_rule.detection_type == AlertRuleDetectionType.STATIC
 
+    @with_feature("organizations:anomaly-detection-alerts")
+    @patch(
+        "sentry.seer.anomaly_detection.store_data.seer_anomaly_detection_connection_pool.urlopen"
+    )
+    def test_update_dynamic_alerts(self, mock_seer_request):
+        mock_seer_request.return_value = HTTPResponse(status=200)
+
+        dynamic_rule = self.create_alert_rule(
+            sensitivity=AlertRuleSensitivity.HIGH,
+            seasonality=AlertRuleSeasonality.AUTO,
+            time_window=60,
+            detection_type=AlertRuleDetectionType.DYNAMIC,
+        )
+        assert mock_seer_request.call_count == 1
+        mock_seer_request.reset_mock()
+        update_alert_rule(dynamic_rule, time_window=30)
```

Comment:
Just to make each bit clear, can we leave some notes on what we're updating on the alert rules, and whether we expect/don't expect them to re-query seer?

Reply:
it seems redundant to me to do
```
# update time_window
update_alert_rule(dynamic_rule, time_window=30)
```

Reply:
For the other updates though it's not as useless?

it's difficult to see exactly what's being updated where.
Adding notes at least helps visually differentiate where each "test" is happening.
alternatively we could also split each of these into their own separate test